modules/rubiconBidAdapter.js

```diff
@@ -136,17 +137,106 @@ var sizeMap = {
   578: '980x552',
   580: '505x656'
 };
+
 _each(sizeMap, (item, key) => sizeMap[item] = key);
 
+export const converter = ortbConverter({
+  request(buildRequest, imps, bidderRequest, context) {
+    const {bidRequests} = context;
+    const data = buildRequest(imps, bidderRequest, context);
+    data.cur = ['USD'];
+    data.test = config.getConfig('debug') ? 1 : 0;
+    deepSetValue(data, 'ext.prebid.cache', {
+      vastxml: {
+        returnCreative: rubiConf.returnVast === true
+      }
+    });
+
+    deepSetValue(data, 'ext.prebid.bidders', {
+      rubicon: {
+        integration: rubiConf.int_type || DEFAULT_PBS_INTEGRATION,
+      }
+    });
+
+    deepSetValue(data, 'ext.prebid.targeting.pricegranularity', getPriceGranularity(config));
+
+    let modules = (getGlobal()).installedModules;
+    if (modules && (!modules.length || modules.indexOf('rubiconAnalyticsAdapter') !== -1)) {
+      deepSetValue(data, 'ext.prebid.analytics', {'rubicon': {'client-analytics': true}});
+    }
+
+    addOrtbFirstPartyData(data, bidRequests);
+
+    delete data?.ext?.prebid?.storedrequest;
+
+    // floors
+    if (rubiConf.disableFloors === true) {
+      delete data.ext.prebid.floors;
+    }
+
+    // If the price floors module is active, then we need to signal to PBS! If floorData obj is present is best way to check
+    const haveFloorDataBidRequests = bidRequests.filter(bidRequest => typeof bidRequest.floorData === 'object');
+    if (haveFloorDataBidRequests.length > 0) {
+      data.ext.prebid.floors = { enabled: false };
+    }
+    // multibid
+    const multibid = config.getConfig('multibid');
```

Comment:
this, and several other things that are (I believe) coming from the original PBS implementation, should already be provided by the library (because you are using `pbsExtensions`). See [multibid](https://github.com/prebid/Prebid.js/blob/db6624c1ec61474576ec0afd26e90e1a708fa9f4/modules/multibid/index.js#L244), [`ext.prebid.floors.enabled`](https://github.com/prebid/Prebid.js/blob/db6624c1ec61474576ec0afd26e90e1a708fa9f4/modules/priceFloors.js#L815), [mediatype from ext.prebid.type](https://github.com/prebid/Prebid.js/blob/db6624c1ec61474576ec0afd26e90e1a708fa9f4/libraries/pbsExtensions/processors/mediaType.js#L17), and others.

I'll note that `ext.prebid.targeting.pricegranularity` was also part of the original PBS adapter but I did _not_ port it over to the library because we determined it was not useful for JS (my understanding is it's an app/amp/mobile feature, needed only when PBS is responsible for targeting). It's likely that you also do not need it here.

Reply:
Thanks for the inputs. I could remove code for multibid and for ext.prebid.type. 

I could not remove floors.enabled part because some (legacy) tests are failing, so i'll do another investigation about it, but it will require more time. It will probably go in another PR